### ISSUE TITLE
CONFIG FILE: Submodule initialization is broken when cloning with --recurse-submodules option 

### ISSUE THREAD

**Issue by altimumdelta** [git-for-windows/git#2125]
- [x] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

### Setup

 - Which version of Git for Windows are you using? Is it 32-bit or 64-bit?

``` 
git version 2.20.1.windows.1
cpu: x86_64
built from commit: 7c9fbc07db0e2939b36095df45864b8cda19b64f                                                    sizeof-long: 4
sizeof-size_t: 8
```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

```
Microsoft Windows [Version 10.0.14393] 64-bit
```

 - What options did you set as part of the installation? Or did you choose the
   defaults?

```
Editor Option: CustomEditor
Custom Editor Path: C:\Program Files (x86)\Programmer's Notepad\pn.exe
Path Option: Cmd
SSH Option: OpenSSH
CURL Option: OpenSSL
CRLF Option: CRLFAlways
Bash Terminal Option: MinTTY
Performance Tweaks FSCache: Enabled
Use Credential Manager: Enabled
Enable Symlinks: Disabled
```

 - Any other interesting things about your environment that might be related
   to the issue you're seeing?

``` no ```

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

``` Bash ```

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

```
 git clone -b master --single-branch --recurse-submodules --origin upstream https://github.com/orgusername/reponame.git
```
 - What did you expect to occur after running these commands?

```
 [submodule "External/dependency"]
	active = true
	url = https://github.com/orgusername/externaldep.git
```

 - What actually happened instead?

```
[submodule]
	active = .
[submodule "External/dependencylibs"]
	url = https://github.com/orgusername/externaldeplibs.git
```

 - If the problem was occurring with a specific repository, can you provide the
   URL to that repository to help us with testing?

** insert URL here **

It works fine when doing it normally after cloning, which is cloning without `--recurse-submodules`, doing a fetch with `--rescurse-submodules`, `git submodule init` then properly creates the submodule entry in the config file.

I actually now realized this is just GitForWindows, this might probably be an upstream git core bug, would it still be okay? (forwarded?)

I read multiple release notes of core git, I don't beleive I noticed any similar mention of this so I think it should be safe to assume the newer versions also have this problem.

**Comment by dscho** [comment]
> ` git clone -b master --single-branch --recurse-submodules --origin upstream https://github.com/orgusername/reponame.git`

This is no easy-to-call **[MCVE](https://stackoverflow.com/help/mcve). Please do not try to offload the entire burden of reproducing this bug onto others, as you are already clearly expecting others to fix the bug.

What you *can* do in order to help others help you is to create a minimal example, i.e. a submodule and a superproject, and put them up on GitHub, as personal repositories. Then at least it would be easier to get started to help you, e.g. by trying the MCVE on Linux to see whether this is a Windows-specific issue.

**Comment by altimumdelta** [comment]
Sure, I can always do a MCVE later, I've ran out of time. As the workaround, to me at least as I manually edit configs to make sure, is already known and very easy, so it's not that critical, therefore there's wasn't a rush on my part.

This also first happened much earlier to me, I did get an error message down the line when using git, I fixed it at the time as well, but I only reported it now after noticing it happen again several months later, so it probably also was happening with the previous version.

**Comment by dscho** [comment]
> And yes I've only reported the problem, not necessairly that I would be the one fixing it, infact if I would be the one fixing it I wouldn't bother making an issue report about it in the first place.

@altimumdelta Please do realize that Git for Windows is used by many, many more users than there are people who help fixing bugs. The latest version, which was released not even a month ago, was already downloaded almost 1.5 *million* times: https://www.somsubhra.com/github-release-stats/?username=git-for-windows&repository=git. Apart from the occasional contributor dropping in, I am the only person taking care of this bug tracker.

If users report bugs with incomplete information, expecting me to hunt down the details and chase magic dragons, like you did, it simply does not scale.

I *want* to help you. But I cannot spend dozens of hours a day *just* trying to reproduce reported problems. I *have* to treat you as an adult who can provide more information, and precise steps (preferably in the form of a script) to reproduce the problem you reported. If you cannot even give me such an MCVE, you basically double the time it would take me to help you, and there are simply not enough hours in the day.

Can you at least have a *little* compassion for this overworked maintainer?

**Comment by altimumdelta** [comment]
Such circumstances are completely understandable, infact it wasn't my intention to provide a half-done report, I was simply assuming it may have not been that strict with this feeling straight-forward to me as it did, that wrong assumption aside I am more than ready to provide anything additional to the issue, however I did not know myself more details until I do more testing. The initial report is already a from-scratch test with actual commands that have been run at the time of writing it. 

That said the common stackoverflow MCVE example is not the most optimal one for all types of issue reporting, it's generalized toward source code programming which is not the area people come from when reporting such issues including myself in this case as I do not have any git build environment setup at this time, it does also not explain what the final report format should be, the term "Example" is not defined at all therfore I have simply followed the issue report template provided here by default. While it is clear that images shouldn't be used, it is still recommended that different projects should have their own version of MCVE explained more clearly.

**Comment by dscho** [comment]
> [...] it is still a good idea that different projects should have their own version of MCVE explained more clearly and specific to the type of environment.

It's a lot of work, in addition to all the rest of the work to keep Git for Windows afloat, to come up with a "Git for Windows-flavored MCVE".

PRs welcome!

**Comment by altimumdelta** [comment]
Someone tested this on archlinux with the latest version, the result is the same:

```
git clone --depth=1 --recurse-submodules --origin upstream https://github.com/git-for-windows/git;
git -C git config --get submodule.active
expected true, got .
```

Therefore there's not much to do around here other than to close it off and label as upstream-bug. I'll be posting all further information to the official git mailing list.

**Comment by dscho** [comment]
>  I'll be posting all further information to the official git mailing list.

Cool! Please do make sure that your mail does not contain HTML, not even as an alternative part, as that would be rejected flat-out by [the Git mailing list](mailto:git@vger.kernel.org).

**Comment by altimumdelta** [comment]
Unfortunately, this happened to be an big misunderstanding between the difference of the "active" parameter in the config file, looking deeply at the docs it appears these are two kinds of "active" parameters, when active = . it's about a pathspec, but active = true it's not a pathspec it's, even if it should be obvious it apparently wasn't. 

The errors I got at the time were not during the cloning or submodule registering process, but after running other commands, I cannot recreate the error messages again and I forgot exactly what I did , so I can only assume those were coincidental false positives.

The only problem here may be inoptimal names/labels of these parameters, but that's another story now, and it would be an idea for upstream anyway.

It has also confused the tester and we made too early conclusions during the process, but I take full responsibility for this false positive, and I apologize for the MCVE arguments as well.

The only thing left to do is replace these labels with an invalid label, or what is appropriate.
